### ISSUE TITLE
Scenario: nested but non-indexed routes

### ISSUE THREAD

**Issue by tomgilder** [tomgilder/routemaster#76]
There might be a scenario where you'd want an embedded nested navigator with a stack of pages, but not using a tab bar or other indexed route.

For instance, two stack next navigators visually next to each other.

**Comment by tomgilder** [comment]
The kind of example scenario I'm thinking of here is an iPad-style app with a sidebar and main view. 

You can already achieve this with Routemaster by, for instance, pushing into the side navigator via `Navigator.push`; but it won't be using the routing engine and won't change the URL.

For URL support this gets complicated - because the URL is going to have to record two different paths, maybe something like `/home/[main/one:sidebar/one]`.

The routing setup could look something like this:

```dart
'/': (_) => NestedPage(
      paths: ['/home/main', '/home/sidebar'],
      child: SplitScreenPage(),
    )
```

...similar to tab pages, but with no index. Thoughts very welcome.

And if anyone knows of another package in any language/framework that does this I'd love to see it!

**Comment by rydmike** [comment]
I thought that might be the case too, and it is a bit related to my use case where I am testing and trying out routemaster together with my still unpublished Flexfold package (https://rydmike.com/flexfold). A demo of an older version of it is live here: https://rydmike.com/demoflexfold. 

The live demo on purpose has no web navigation. Flexfold package is intended to be totally agnostic to how you navigate. Its setup work a bit like the Flutter SDK navigation rail. You just define destinations, but you have to take care of navigating to them yourself in whatever manner you like.

I will describe my use cases and question/issues I have faced so far in separate question/issue post when trying to add routemaster to the demo app that showcases Flexfold.

**Comment by julienlebren** [comment]
It would be a good feature!

Another use case example: A form flow in multiple steps which open in a fullScreenDialog, which is very common nowadays in mobile apps: the form is divided in steps with something like one field per view. But at any time, he should be able to cancel the form. In this case, if the user is at the step 3 of the form, the user is not popped inside the current Navigator but the main navogator is popped, so the view opened in fullScreenDialog is closed.

Not sure if I'm clear 😅

**Comment by tomgilder** [comment]
Initial version of nested routes using `StackPage` is available in 0.10.0-dev1.

See the mobile_app example app for an example of it:

```dart
      '/stack': (_) => StackPage(
            pageBuilder: (child) => BottomSheetPage(child: child),
            child: StackBottomSheetContents(),
            defaultPath: 'one',
          ),

      '/stack/one': (_) => MaterialPage(child: StackPageOne()),
      '/stack/one/two': (_) => MaterialPage(child: StackPageTwo()),
```
